resources/aws_ec2.go

```diff
@@ -708,3 +737,42 @@ func (obj *AwsEc2Res) snsPostHandler(w http.ResponseWriter, req *http.Request) {
 		log.Printf("%s: Post: %s", obj, string(post))
 	}
 }
+
+// snsMakeTopic creates a topic on aws sns.
+func (obj *AwsEc2Res) snsMakeTopic() (string, error) {
+	// make topic
+	topicInput := &sns.CreateTopicInput{
+		Name: aws.String(SnsTopicName),
+	}
+	topic, err := obj.snsClient.CreateTopic(topicInput)
+	if err != nil {
+		return "", err
+	}
+	log.Printf("%s: Created SNS Topic", obj)
+
+	var topicArn string
+	if topic.TopicArn != nil {
+		topicArn = *topic.TopicArn
+	}
+	return topicArn, nil
+}
+
+// Close cleans up when we're done. This is needed to delete some of the AWS
+// objects created for the SNS endpoint.
+func (obj *AwsEc2Res) Close() error {
+	var errList error
+	if obj.WatchListenAddr != "" {
+		// delete the topic
+		dtInput := &sns.DeleteTopicInput{
```

Comment:
one nit about asymmetry: you have a `snsMakeTopic` to do the making, where as the inverse to that is just inline in the close method. It's not a hard rule, but usually both would either be inline, or both would be in methods. I think the latter might be appropriate.

2nd thing: suppose `Init` runs, but returns early with `"error creating sns session"`, and then `Close` gets called-- will running this DeleteTopic stuff error because we didn't have a proper session? Probably good to make sure that code is safe in both scenarios. All the better for it to have it's own method to do so and grow in.

Reply:
will do.

Reply:
It will error, but if there was an error creating the session in Init, it would error anyway. I'm not sure there's any benefit in creating it to delete objects that, by definition, can't have been made in the first place.

Reply:
What I mean was, `Close` should not unnecessarily error trying to delete something via a session that does not exist because it was not created in `Init`. The solution isn't to try and create it in `Close`-- the solution is to skip over the deletion in `Close` if the session is nil.